tests/ui/cfg/field_sneaky.rs

```diff
@@ -0,0 +1,19 @@
+// compile-fail
+// aux-build:sneaky_macro.rs
+
+#![feature(optin_builtin_traits)]
+#![feature(trivial_bounds)]
+
+#[macro_use]
+extern crate sneaky_macro;
+
+use pin_project::pin_project;
+
+#[pin_project] //~ ERROR pattern does not mention field `__field`
+#[add_pinned_field]
```

Comment:
I have never seen a crate actually attacked like this, but it should be preferable to be able to detect it.